CollapseLauncher/Classes/Helper/HttpClientBuilder.Dns.cs

```diff
@@ -0,0 +1,529 @@
+﻿using Hi3Helper;
+using Hi3Helper.Win32.Native.Structs.Dns.RecordDataType;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+using TurnerSoftware.DinoDNS;
+using TurnerSoftware.DinoDNS.Protocol;
+using Dns = Hi3Helper.Win32.ManagedTools.Dns;
+// ReSharper disable CheckNamespace
+// ReSharper disable StaticMemberInGenericType
+// ReSharper disable IdentifierTypo
+// ReSharper disable CommentTypo
+
+#nullable enable
+namespace CollapseLauncher.Helper
+{
+    public partial class HttpClientBuilder<THandler>
+    {
+        private readonly struct CachedDnsResolveTtlInfo
+        {
+            public CachedDnsResolveTtlInfo(double ttl)
+            {
+                ValidUntil = DateTimeOffset.Now.AddSeconds(ttl);
+            }
+
+            public readonly DateTimeOffset ValidUntil;
+        }
+
+        internal static readonly Dictionary<string, IPAddress[]> DnsServerTemplate = new(StringComparer.OrdinalIgnoreCase)
```

Comment:
[Suggestion] Add quad9 resolver
https://quad9.net/service/service-addresses-and-features 

Using insecure so it queries a bit faster as we don't verify DNSSEC anyway